specifyweb/frontend/js_src/lib/components/Forms/Save.tsx

```diff
@@ -214,27 +221,82 @@ export function SaveButton<SCHEMA extends AnySchema = AnySchema>({
     </ButtonComponent>
   );
 
+  const [carryForwardAmount, setCarryForwardAmount] = React.useState<number>(1);
+
   return (
     <>
       {typeof handleAdd === 'function' && canCreate ? (
         <>
-          {showClone &&
-            copyButton(
-              formsText.clone(),
-              formsText.cloneDescription(),
-              async () => resource.clone(true)
-            )}
+          {resource.specifyTable.name === 'CollectionObject' &&
+          (isInRecordSet === false || isInRecordSet === undefined) &&
+          isSaveDisabled &&
+          showCarry &&
+          showBulkCarry ? (
+            <Label.Inline>
+              <Input.Generic
+                type="number"
+                value={carryForwardAmount}
+                onValueChange={(value): void =>
+                  carryForwardAmount === undefined
+                    ? setCarryForwardAmount(1)
+                    : setCarryForwardAmount(Number.parseInt(value))
+                }
+              />
+            </Label.Inline>
+          ) : null}
           {showCarry &&
             copyButton(
               formsText.carryForward(),
               formsText.carryForwardDescription(),
-              async () => resource.clone(false)
+              resource.specifyTable.name === 'CollectionObject' &&
+                carryForwardAmount > 2
+                ? async () => {
+                    const formatter =
+                      tables.CollectionObject.strictGetLiteralField(
+                        'catalogNumber'
+                      ).getUiFormatter()!;
+                    const wildCard = formatter.valueOrWild();
+
+                    const clonePromises = Array.from(
+                      { length: carryForwardAmount },
+                      async () => {
+                        const clonedResource = await resource.clone(false);
+                        await clonedResource.set(
+                          'catalogNumber',
+                          wildCard as never
+                        );
+                        return clonedResource;
+                      }
+                    );
+
+                    const clones = await Promise.all(clonePromises);
+
+                    const backendClones = await ajax<
```

Comment:
the code is clonning a resource x times and then sending that as JSON to the back-end
do we have the capability to do cloning on the back-end? i.e can we just send a base record and a number to the back-end and let it do the rest? would be more bandwidth efficient 🙂

Reply:
@acwhite211

Reply:
Ya, I had a similar thought.  The request body size would get pretty massive when someone does a carry-forward with 100 plus copies.  I added an api `/api/specify/bulk/collectionobject/99/` that calls a collection_dispatch_bulk_copy function.  The api input is the same as the colleection_dispatch, you just add a number to the end of the api request for the amount of copies.  When could change it to be `/api/specify/collectionobject?copies=99`, that might be more clear, let me know what you think.

Reply:
So, we decided that while the new API works in reducing the request body size, the front-end currently still needs the response body to return all the newly created records to create a record set.  This defeats the purpose of the new `collection_dispatch_bulk_copy` function.  So, we'll keep the API for now, but we'll need to wait to call it until the front-end can handle not receiving all the created records in the response.  We could implement it now only if we are for sure that we don't need the front-end to create a record set of the new data.

Reply:
What is stopping us from returning the same response (an array of the created resources) from the `new collection_dispatch_bulk_copy` endpoint?

We are still calling `post_resource` for each record we create, which returns the created object. 

https://github.com/specify/specify7/blob/bc06abf437cae5fd0debb3838e3c0a64b9ed184e/specifyweb/specify/api.py#L289-L300

Was there somehow trouble regarding serializing the objects? It should be exactly the same as the current endpoint we are using: 

https://github.com/specify/specify7/blob/bc06abf437cae5fd0debb3838e3c0a64b9ed184e/specifyweb/specify/api.py#L268-L278